### DISCUSSION TITLE
Saving new related item

### DISCUSSION THREAD

**Discussion by rgieseke** [directus/directus#11150] [Archive]
Is there a way to directly save a newly created item of a related table (e.g. from a many to one relation), when it shows in an overlay panel after clicking on the + button or "Create New"?

Right now it is only saved when saving the top-level item.

**Comment by joselcvarela** [comment]
Not currently.
Can you explain your use case so we could consider it?
Why does not the top label works for you?

**Reply by rgieseke** [reply]
Sure, it's related to a structure as discussed in https://github.com/directus/directus/discussions/10986

We have added an "Apply defaults" button and can now do the following:

1. Click Add author (serial_article_author)
2. Click create new person (persons)
3. Click Add Affiliation (person_affiliations)
4. Click create new organization (organizations), enter data, click "save" (doesn't actually save yet so we don't have an id yet)
5. Click "save" on affiliation (person_affiliations, doesn't actually save yet)
6. Click "save" on person (persons, doesn't actually save yet)
7. Click "apply defaults" -> serial_article_affiliations gets filled with unsaved entries from person_affiliations which contain unsaved entries from organizations 

After saving on the root level we will have all newly created organizations twice

As far as i understood there can be 3 cases for an edit:
```
* The value of a many to one can be one of three things: A primary key (number/string), a nested
 * object of edits (including primary key = editing existing) or an object with new values (no
 * primary key)
 ```
 https://github.com/directus/directus/blob/5e7fa633b0fbb5d15b78a69a9ef0f6fc86ec5698/app/src/interfaces/select-dropdown-m2o/select-dropdown-m2o.vue#L108
 
 I think it could in general be useful to create the new related item right away and have the id.

Maybe this could be an option similar to "save and stay".

**Comment by rgieseke** [comment]
To maybe better explain the problem:
Consider articles, which have (multiple) authors (a table of persons) who each have an affiliation (a table of organisations).

After creating an article, one adds an author and to that author an affiliation is added, a new entry in the table of organisations.
When a 2nd author is added to the article with the same affiliation it's not possible to reference the previously created organisation because the article has not yet been saved (potentially leading to a duplicate entry).

**Reply by andrenarchy** [reply]
This is indeed a very important issue for us right now.

One solution would be to set in the parent's field config (for that relationship) how the item is saved: 
* Save immediately
* Save with parent

If both are available then the user might even choose (similar to "Save+stay" on the top level). What do you think?

**Comment by ImaCrea** [comment]
Hello, we've run last week a few (7 people) user testing session on a directus instance I'm working on for a client and I've seen that situation appear in almost all sessions. It's not when user creates a related item but when she edits it. As it's a very similar usecase, on a UX point of view, I believe sharing this would help :

- User opens an item
- Click on an element from a relational field
- Enjoy the fact she can edit the info of that related item right away
- Click on save button
- And then thinks it's done. 
- So when asked to check another item, she simply leaves that view and the popup saying there are modifications not saved is displayed. She doesn't understand what it's about and in all cases where this happened, she clicks on "cancel edits" without realizing she just canceled what she edited in the related item. At this point, she could also have closed the tab.

When explained after that why the popup, she understood but there's no safe way to know if that misunderstanding may happen again.
